qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py

```diff
@@ -907,6 +908,64 @@ def autorange_once(self) -> None:
         self.write('SENSe:VOLTage:DC:RANGe:AUTO ONCE')
         self.range.get()
 
+    def increase_range(
+        self,
+        range_value: Optional[Union[float, int]] = None,
+        increase_by: int = 1
+    ) -> None:
+        """
+        Increases the voltage range by a certain amount with default of 1.
+        If limit is reached, the max range is used.
+
+        Args:
+            range_value : The desired voltage range needed.  Expressed by power
+                of 10^x range from -3 to 10
+            increase_by: How much to increase range by, default behavior
+                 is by a step of one.
+
+        """
+        if increase_by < 1:
+            raise ValueError("The steps must be increasing in value")
+
+        current_range = self.range.get()
```

Comment:
Lets move this call to get inside the if statement. That way we can avoid calling get if a user has supplied the range

Reply:
I used an if-else for the statement.  Is that ok?